### PR TITLE
Flasher tab: remember last user selected radio protocol

### PR DESCRIPTION
**Pull request by limonspb**
Flasher tab.
Stores the last radio protocol selected by the user into localStorage.
Sets radio protocol dropdown to the last selected value (if value is present in the list, otherwise shows default value)
Update: everything is ignored if buildkey is present = radio protocol is loaded from buildkey.

### CODE REVIEW

File: src/js/tabs/firmware_flasher.js
Comment (blckmn) [inline src/js/tabs/firmware_flasher.js:423]: @limonspb can you check this doesn't contradict the setting of this value based on the build key ? i.e. its ok here, so long as the application of the build key query overrides it.

Comment (limonspb) [reply src/js/tabs/firmware_flasher.js:429]: @blckmn good note.
I have updated PR. Now if build key exists (length === 32) then it ignores protocol from localStorage.

**Comment by blckmn** [comment]
AUTOMERGE: (FAIL)
 - github identifies PR as mergeable -> FAIL
 - assigned to a milestone -> FAIL
 - cooling off period lapsed -> FAIL
 - commit count less or equal to three -> PASS
 - `Don't merge` label NOT found -> PASS
 - at least one `RN:` label found -> FAIL
 - `Tested` label found -> FAIL
 - assigned to an approver -> FAIL
 - approver count at least three -> FAIL

**Comment by haslinghuis** [comment]
This should work for 4.4.3+: https://github.com/betaflight/betaflight/pull/12876

**Review by blckmn — APPROVED** [review]

**Comment by haslinghuis** [comment]
@limonspb as this code won't be needed for 4.5.0 anymore would like some semver conditions added.

This PR only works if using one specific type of receiver. In a mixed environment this only adds extra maintenance.

Shame there is no #IF_IAM_NOT_LAZY possibility :stuck_out_tongue:

**Comment by limonspb** [comment]
> @limonspb as this code won't be needed for 4.5.0 anymore would like some semver conditions added - but actually we should reject the change:
> 
> * it would already be fixed in 4.4.3 in #12876
> * PR only works if using one specific type of receiver.
> * in a mixed environment this only adds extra maintenance.
> 
> Shame there is no `#USE_IF_IAM_NOT_LAZY` macro 😛 🙈

lots of people will be upgrading from 4.2 4.3, from 4.4.0, from 4.4.1, and from 4.4.2. Heck, there are people still on 4.0 lol.
This PR does not change anything, if you are already at 4.5 or 4.4.3, but it helps to others who upgrading from older versions. Thats the only purpose.

**Comment by haslinghuis** [comment]
> but it helps to others who upgrading from older versions with multiple drones.

assuming they all have the same receivers. Soon everyone will at least be flashing 4.4.3 :)

**Comment by limonspb** [comment]
> > but it helps to others who upgrading from older versions with multiple drones.
> 
> assuming they all have the same receivers. Soon everyone will at least be flashing 4.4.3 :)
> 
> @McGiverGim what's your opinion?
> 
> This PR assumes configurator will be released before 4.5.0. Sure have no problem with that but it should be coupled at least with 4.4.3 and introduce a feature freeze (unless its 10.9.1). Rather have some more time to test nightly builds for now.
> 
> 4.5.0 should not take too long as products are already out there.

thats overestimation that everyone will be on 4.4.3 :)

**Comment by McGiverGim** [comment]
> @McGiverGim what's your opinion?

I think is useful. In my case, for example, I'm on FrSky FPORT, that is not selected automatically. I have few drones, but if someone has some dotzens then it's useful. The same can be done for some "extra" flags, like MAG or similar.

**Comment by haslinghuis** [comment]
@limonspb please rebase so we can merge this.

**Review by haslinghuis — APPROVED** [review]

**Pull request merged**
